### ISSUE TITLE
near-operation-file preset should allow for types to be in external package

### ISSUE THREAD

**Issue by willtrking** [dotansimha/graphql-code-generator#2912]
Internally, we package our generated schema types independently of our generated operations and react components.

We'd like to use the `near-operation-file` preset, and it would be nice if it allowed for types to be imported from an external package. Currently doing something like the following:

```
overwrite: true
schema: './node_modules/@internal/gql/dist/schema/*.graphql'
documents: './graphql/**/*.gql.ts'
generates:
  /:
    preset: near-operation-file
    presetConfig:
      extension: .generated.tsx
      baseTypesPath: '@internal/gql'
      folder: generated
    plugins:
      - 'typescript-operations'
      - 'typescript-react-apollo'
    config:
      withHooks: true
      withHOC: false
      reactApolloVersion: 3
```

Ends up with imports as follows

`import * as Types from '../../../../../../../@internal/gql';`

I propose adding a new option, `baseTypesPackage` to allow for this use case. Adding a new option rather then re-purposing the existing `baseTypesPath` would ensure no existing uses of that option break.

I'm happy to submit a PR for this.

**Comment by ardatan** [comment]
Could you try with the following canary version by adding `baseTypesPath: '~@internal/gql'?
`1.8.4-alpha-11f86a7d.48+11f86a7d`

**Comment by willtrking** [comment]
Worked like a charm @ardatan, thank you!

**Comment by kylecannon** [comment]
@ardatan any idea when you would be releasing this as latest on npm?

**Comment by dotansimha** [comment]
Fixed in 1.9.0

**Comment by KnisterPeter** [comment]
@dotansimha This is missing in the docs. It is really helpful if used in a monorepository as well. :rocket:
